frontend/public/components/dashboards-page/overview-dashboard/top-consumers-card.tsx

```diff
@@ -0,0 +1,137 @@
+import * as React from 'react';
+import * as _ from 'lodash-es';
+
+import * as plugins from '../../../plugins';
+import {
+  DashboardCard,
+  DashboardCardBody,
+  DashboardCardHeader,
+  DashboardCardTitle,
+} from '../../dashboard/dashboard-card';
+import { ConsumersBody, ConsumersFilter, metricTypeMap } from '../../dashboard/top-consumers-card';
+import {
+  NODES,
+  PODS,
+} from '../../dashboard/top-consumers-card/strings';
+import { DashboardItemProps, withDashboardResources } from '../with-dashboard-resources';
+import { Dropdown } from '../../utils';
+import { OverviewQuery, topConsumersQueries } from './queries';
+import { getInstantVectorStats } from '../../graphs/utils';
+import { BarChart } from '../../graphs/bar';
+import { DataPoint } from '../../graphs';
+import { MetricType } from '../../dashboard/top-consumers-card/metric-type';
+
+const topConsumersQueryMap: TopConsumersMap = {
+  [PODS]: {
+    metric: 'pod_name',
+    queries: {
+      [MetricType.CPU]: topConsumersQueries[OverviewQuery.PODS_BY_CPU],
+      [MetricType.MEMORY]: topConsumersQueries[OverviewQuery.PODS_BY_MEMORY],
+      [MetricType.STORAGE]: topConsumersQueries[OverviewQuery.PODS_BY_STORAGE],
+    },
+  },
+  [NODES]: {
+    metric: 'node',
+    queries: {
+      [MetricType.CPU]: topConsumersQueries[OverviewQuery.NODES_BY_CPU],
+      [MetricType.MEMORY]: topConsumersQueries[OverviewQuery.NODES_BY_MEMORY],
+      [MetricType.STORAGE]: topConsumersQueries[OverviewQuery.NODES_BY_STORAGE],
+      [MetricType.NETWORK]: topConsumersQueries[OverviewQuery.NODES_BY_NETWORK],
+    },
+  },
+};
+
+const getTopConsumersQueries = (): TopConsumersMap => {
+  const topConsumers = {...topConsumersQueryMap};
+
+  const pluginItems = plugins.registry.getDashboardsOverviewTopConsumerItems();
+  pluginItems.forEach(pluginItem => {
+    if (topConsumers[pluginItem.properties.name]) {
```

Comment:
This can be simplified

```ts
if (!topConsumers[pluginItem.properties.name]) {
  topConsumers[pluginItem.properties.name] = {
    metric: pluginItem.properties.metric,
    queries: pluginItem.properties.queries,
    mutator: pluginItem.properties.mutator,
  };
}
```

Reply:
done